src/System.Linq/tests/LegacyTests/GroupByTests.cs

```diff
@@ -1151,5 +1151,23 @@ public void Test()
                 Assert.Equal(0, Main());
             }
         }
+        
+        [Fact]
+        public void GroupingToArray()
+        {
+            Record[] source = new Record[]
+            {
+                new Record{ Name = "Tim", Score = 55 },
+                new Record{ Name = "Chris", Score = 49 },
+                new Record{ Name = "Robert", Score = -100 },
+                new Record{ Name = "Chris", Score = 24 },
+                new Record{ Name = "Prakash", Score = 9 },
+                new Record{ Name = "Tim", Score = 25 }
+            };
+            
+            IGrouping<string, Record>[] groupedArray = source.GroupBy(r => r.Name).ToArray();
+            Assert.Equal(4, groupedArray.Length);
+            Assert.Equal(source.GroupBy(r => r.Name), groupedArray);
```

Comment:
NP. Though actually a lot of the legacy linq tests do indeed test things against themselves, which can seem rather strange.